msrest/serialization.py

```diff
@@ -757,6 +757,8 @@ def __call__(self, target_obj, response_data):
                     working_data = working_data.get(working_key, data)
                     key = '.'.join(dict_keys[1:])
 
+                if key not in working_data:
+                    continue
                 raw_value = working_data.get(key)
```

Comment:
If you expect the key to be there the majority of the time you can simplify this to:
```python
try:
    raw_value = working_data[key]
except KeyError:
    continue
```